src/core/observables/ProfileObservable.hpp

```diff
@@ -34,25 +34,37 @@ namespace Observables {
 
 /** Cartesian profile observable */
 class ProfileObservable : virtual public Observable {
+protected:
+  /** Range of the profile edges. */
+  const std::array<std::pair<double, double>, 3> limits;
+  /** Number of bins for each coordinate. */
+  const std::array<size_t, 3> n_bins;
```

Comment:
... too.

There reason is that this limits how the type can be used, which breaks the separation, but has no actual benefit. Also `protected` is usually no a great sign, design-wise. This is seven a core guideline.